SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityDeployChecker.java

```diff
@@ -346,15 +360,17 @@ private SingularityDeployResult fromLbState(DeployState state, SingularityLoadBa
   }
 
   private boolean shouldCheckLbState(final SingularityPendingDeploy pendingDeploy) {
-    return pendingDeploy.getLastLoadBalancerUpdate().isPresent() && (pendingDeploy.getLastLoadBalancerUpdate().get().getLoadBalancerState() != BaragonRequestState.UNKNOWN);
+    return pendingDeploy.getLastLoadBalancerUpdate().isPresent()
+        && getLoadBalancerRequestId(pendingDeploy).getId().equals(pendingDeploy.getLastLoadBalancerUpdate().get().getLoadBalancerRequestId().getId())
```

Comment:
can you clarify what this comparison is for? (i assume related to the incremental deploy, but having a hard time understanding it)